### DISCUSSION TITLE
Syntax for multi-function food processor "commands"

### DISCUSSION THREAD

**Discussion by titonobre** [cooklang/spec#83] [Ideas]
Hi.
First of all, awesome project here! And great ideas in the discussions. 

Some of my recipes were developed for a multi-function food processor, I my case a [Thermomix/Bimby](https://en.wikipedia.org/wiki/Thermomix) but most recipes can be prepared in similar appliances with the same functions.

These recipes have some instructions that are specific for these types of appliances, like speed, temperature, blade direction and duration of each step.

Some examples, of what we can find in books or online:

![image](https://user-images.githubusercontent.com/7109082/158901984-d3818daf-e4df-4e1f-9b5b-6f2762711cf0.png)
[here](https://www.recipecommunity.com.au/main-dishes-meat-recipes/bourbon-ribs/brxygvj3-b9792-902138-cfcd2-k842cpd5), meaning 5 seconds at speed 7

![image](https://user-images.githubusercontent.com/7109082/158901753-2ed177f9-cb1a-443b-ab94-6f683b71fc8f.png)
[here](https://www.recipecommunity.com.au/main-dishes-meat-recipes/hearty-oxtail-stew/90djfhln-9d870-482951-cfcd2-p0tjejhl), meaning 60 minutes at 100°C with counter-clockwise gentle stirring.

Would it be possible to have this in Cooklang?

Something like:

```cook
~{5%seconds}/${7}
```

or (with temperature idea from #67 )

```cook
~{60%minutes}/@{100%C}/${-stirring}
```

The only thing missing is the syntax for the speed, for which I used `${}` with the `-` for reverse direction.

**Comment by gvlx** [comment]
Extending from https://github.com/cooklang/spec/discussions/67#discussioncomment-15886944 we could use `#blender{20%m}{50%C}{-stirring}`.

The speed syntax could be something in the lines of : 
- nothing or plus | minus for reverse
- stirring|speed (numeric) | turbo
- (other capabilities?)
- ending with %T (which does not seems to be used)

compiler/editor should check for incompatible combinations

**Comment by johnuopini** [comment]
I am in the same situation and this will be especially great if some of these appliances will ever introduce a way to send recipes to them via some API
